crates/sui-config/src/genesis_config.rs

```diff
@@ -126,6 +132,9 @@ impl ValidatorGenesisInfo {
             network_key_pair,
             network_address: utils::new_tcp_network_address(),
             p2p_address: utils::new_udp_network_address(),
+            p2p_listen_address: None,
+            metrics_address: utils::available_local_socket_address(),
```

Comment:
It seems metrics address should be in a separate per-host config, where other validators do not need to know about? Not sure what is the infra for the 2nd option though. cc @longbowlu for validator metadata.

Reply:
cc @bmwill: since you've been working on both genesis tooling and the new monitoring infra, do you have any opinion about the best way to move forward?